syne_tune/optimizer/schedulers/multiobjective/nsga2_searcher.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import numpy as np
+
+from typing import Optional, List, Union, Dict, Any
+
+from syne_tune.optimizer.schedulers.searchers import StochasticSearcher
+from syne_tune.config_space import Domain, Float, Integer, Categorical, FiniteRange
+
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.core.problem import Problem
+from pymoo.core.evaluator import Evaluator
+from pymoo.problems.static import StaticProblem
+from pymoo.core.mixed import (
+    MixedVariableMating,
+    MixedVariableSampling,
+    MixedVariableDuplicateElimination,
+)
+from pymoo.core.variable import Real, Choice
+from pymoo.core.variable import Integer as PyMOOInteger
+
+
+class MultiObjectiveMixedVariableProblem(Problem):
+    def __init__(self, n_obj: int, config_space: Dict[str, Any], **kwargs):
+        vars = {}
+
+        for hp_name, hp in config_space.items():
+            if isinstance(hp, Categorical):
+                vars[hp_name] = Choice(options=hp.categories)
+            elif isinstance(hp, Integer):
+                vars[hp_name] = PyMOOInteger(bounds=(hp.lower, hp.upper))
+            elif isinstance(hp, FiniteRange):
+                vars[hp_name] = PyMOOInteger(bounds=(0, hp.size - 1))
+            elif isinstance(hp, Float):
+                vars[hp_name] = Real(bounds=(hp.lower, hp.upper))
+
+        super().__init__(vars=vars, n_obj=n_obj, n_ieq_constr=0, **kwargs)
+
+
+class NSGA2Searcher(StochasticSearcher):
+    """
+    This is a wrapper around the NSGA-2 [1] implementation of pymoo [2].
+
+    [1] K. Deb, A. Pratap, S. Agarwal, and T. Meyarivan.
```

Comment:
Prefix these lines with "|".

See `syne_tune/optimizer/schedulers/searchers/bayesopt/models/meanstd_acqfunc_impl.py` for example